schema-layer/data-structures/Data.md

```diff
@@ -0,0 +1,62 @@
+`Data` is an advanced layout for representing binary data.
+
+It is flexibible enough to support very small and very large (multi-block) binary data.
+
+```sh
+type Index [Int]
```

Comment:
Is this supposed to just be `type Index Int`, perhaps?

Reply:
It’s correct. The index is a tuple of `[index, length]`. Ideally this would also enforce the length though, so I’m keen to hear any suggestions for improvements.

Reply:
I'm not entirely clear on what "index" means in here. Just a list of parts and their lengths, and if so what is the first part of that tuple for? Couldn't it just be an array of lengths and the index is the zero-based index of that list?

If you want to enforce this as a strict tuple of two things then this is probably what you want:

````
type Index struct {
  index Int
  length Int
} representation tuple
 ```

But the duplication of "index" here could maybe be addressed to make it clearer what this is for?

Reply:
factored out.